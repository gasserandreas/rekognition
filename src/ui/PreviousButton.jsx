import React from 'react';
import styled from 'styled-components';

import { Previous } from 'grommet-icons';

import { Colors } from '../styles';

export const StyledPrevious = styled(Previous)`
  display: block;
  visibility: visible;

  &:hover {
    stroke: ${Colors.ColorsPalette.TextFaded};
    cursor: pointer;
  }
`;

/** Render go previous button as left arrow icon */
const PreviousButton = (props) => <StyledPrevious {...props} />;

export default PreviousButton;
