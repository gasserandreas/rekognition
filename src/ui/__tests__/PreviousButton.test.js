import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PreviousButton, { StyledPrevious } from '../PreviousButton';

// test styling
it('PreviousButton should render correctly', () => {
  const wrapper = shallow(<PreviousButton />);
  expect(toJson(wrapper.dive())).toMatchSnapshot();
});

it('PreviousButton styles should be consistent', () => {
  const wrapper = shallow(<StyledPrevious />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
