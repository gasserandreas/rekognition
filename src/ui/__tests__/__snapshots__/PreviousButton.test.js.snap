// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviousButton should render correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: block;
  visibility: visible;

  &:hover {
    stroke: ",
          "#42526E",
          ";
    cursor: pointer;
  }
",
        ],
      },
      "displayName": "Styled(Previous)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`PreviousButton styles should be consistent 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: block;
  visibility: visible;

  &:hover {
    stroke: ",
          "#42526E",
          ";
    cursor: pointer;
  }
",
        ],
      },
      "displayName": "Styled(Previous)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    }
  }
  forwardedRef={null}
/>
`;
